source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -78,46 +78,57 @@ def get_max_fallback_duration(request, mode, dp_durations={}):
     raise ValueError('unknown mode: {}'.format(mode))
 
 
-def make_direct_path_key(dep_mode, orig_uri, dest_uri, datetime, clockwise, reverse_sections):
-    # datetime is not taken into consideration because we assume that
-    # a direct path from A to B remains the same even the departure time are different
-    return dep_mode, orig_uri, dest_uri, clockwise, reverse_sections
+def make_direct_path_key(dep_mode, orig_uri, dest_uri, is_fallback_at_end,
+                         datetime, is_start_fallback_datetime):
+    '''
+    A, B and mode matters obviously
```

Comment:
A and B does not appear above, it's a bit confusing

and maybe add a bit more general comment first